src/app/components/signin/signin.component.html

```diff
@@ -1,11 +1,17 @@
-<form [formGroup]="form">
-    <h2>Sign in</h2>
+<form [formGroup]="signinForm" (ngSubmit)="onSubmit()">
+     <h2>Sign in</h2>
   
     <label>Email</label><br><br>
     <input type="email" formControlName="email"><br><br>
+    <div *ngIf="signinForm.get('email')!.errors && signinForm.get('email')!.touched">
+        <div *ngIf="signinForm.get('email')?.hasError('required')" [style.color]="textColor" [style.font-size.rem]="errorMsgFontSize">email is required</div>
+    </div>
     <label>Password</label><br><br>
     <input type="password" formControlName="password"><br><br>
+    <div *ngIf="signinForm.get('password')!.errors && signinForm.get('password')!.touched">
+        <div *ngIf="signinForm.get('password')?.hasError('required')" [style.color]="textColor" [style.font-size.rem]="errorMsgFontSize">password is required</div>
+    </div> 
     
-    <button type="submit">Sign in</button>
+    <button type="submit" (click)="onSubmit()">Sign in</button>
```

Comment:
with the onsubmit on the form, you don't need the click event on the button as long as it's type submit

Reply:
no problem just fixing the problem